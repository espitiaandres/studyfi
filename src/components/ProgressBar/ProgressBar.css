.progress {
    border: 1px solid #eee;
    height: 4px;
    border-radius: 2px;
    overflow: hidden;
    width: 300px;
}

.progressBar {
    background-color: #1ED760;
    color: white;
    height: 6px;
}

.durationMenu {
    align-items: center;
    color: white;
    display: flex;
    justify-content: space-evenly;
}

.durationMenuTimes {
    padding: 0px 4px;
}

.seasonStylingAlt {
    background-color: #FFE01B;
    border-color: #FFE01B;
}
