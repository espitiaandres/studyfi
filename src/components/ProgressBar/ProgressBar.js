//
//  ProgressBar.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import moment from 'moment';
import './ProgressBar.css';

const ProgressBar = ({ songCurrentTime, songDuration, seasonStylingAlt }) => {
    const songCurrentTimeMinutesSeconds = moment(songCurrentTime).format("mm:ss");
    const songDurationMinutesSeconds = moment(songDuration).format("mm:ss");

    return (
        <div className="durationMenu">
            <p className="durationMenuTimes">{songCurrentTimeMinutesSeconds}</p>
                <div className="progress">
                    <div className={`progressBar ${seasonStylingAlt}`} style={{
                        width: (songCurrentTime * 100 / songDuration) + '%'
                    }} />
                </div>    
            <p className="durationMenuTimes">{songDurationMinutesSeconds}</p>
        </div>
    )
}

export default ProgressBar;
