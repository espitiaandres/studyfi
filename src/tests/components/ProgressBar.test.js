//
//  ProgressBar.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from '../../components/ProgressBar/ProgressBar';

const progressBarPropsTest = {
    songCurrentTime: 1000,
    songDuration: 10000,
    seasonStylingAlt: "seasonStylingAlt"
}

test('should render ProgressBar component correctly', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ProgressBar component correctly with data', () => {
    const wrapper = shallow(
        <ProgressBar
            songCurrentTime={progressBarPropsTest.songCurrentTime}
            songDuration={progressBarPropsTest.songDuration}
            seasonStylingAlt={progressBarPropsTest.seasonStylingAlt}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
