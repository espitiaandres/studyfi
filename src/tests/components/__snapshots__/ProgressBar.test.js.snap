// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProgressBar component correctly 1`] = `ShallowWrapper {}`;

exports[`should render ProgressBar component correctly with data 1`] = `ShallowWrapper {}`;
